units/event_metajob.go

```diff
@@ -0,0 +1,195 @@
+		return !flags.WebhookNotificationsDisabled
+
+	case event.EmailSubscriberType:
+		return !flags.EmailNotificationsDisabled
+
+	case event.SlackSubscriberType:
+		return !flags.SlackNotificationsDisabled
+
+	default:
+		grip.Alert(message.Fields{
+			"message": "notificationIsEnabled saw unknown subscriber type",
+			"cause":   "programmer error",
+		})
+	}
+
+	return false
+}
+
+type eventMetaJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	q        amboy.Queue
+	events   []event.EventLogEntry
+	flags    *evergreen.ServiceFlags
+}
+
+func makeEventMetaJob() *eventMetaJob {
+	j := &eventMetaJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    eventMetaJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	j.SetPriority(1)
+
+	return j
+}
+
+func NewEventMetaJob(q amboy.Queue, ts string) amboy.Job {
+	j := makeEventMetaJob()
+	j.q = q
+
+	j.SetID(fmt.Sprintf("%s:%s", eventMetaJobName, ts))
+
+	return j
+}
+
+func (j *eventMetaJob) dispatchLoop() error {
+	// TODO: if this is a perf problem, it could be multithreaded. For now,
+	// we just log time
+	startTime := time.Now()
+	logger := event.NewDBEventLogger(event.AllLogCollection)
+	catcher := grip.NewSimpleCatcher()
+	for i := range j.events {
+		var notifications []notification.Notification
+		notifications, err := notification.NotificationsFromEvent(&j.events[i])
+		catcher.Add(err)
+
+		grip.Error(message.WrapError(err, message.Fields{
+			"job_id":     j.ID(),
+			"job":        eventMetaJobName,
+			"source":     "events-processing",
+			"message":    "errors processing triggers for event",
+			"event_id":   j.events[i].ID.Hex(),
+			"event_type": j.events[i].ResourceType,
+		}))
+
+		// TODO: buffered writes after EVG-3062
+		if err = notification.InsertMany(notifications...); err != nil {
+			catcher.Add(err)
+			continue
+		}
+
+		catcher.Add(j.dispatch(notifications))
+		catcher.Add(logger.MarkProcessed(&j.events[i]))
+	}
+	endTime := time.Now()
+	totalDuration := endTime.Sub(startTime)
+
+	grip.Info(message.Fields{
+		"job_id":     j.ID(),
+		"job":        eventMetaJobName,
+		"source":     "events-processing",
+		"message":    "stats",
+		"start_time": startTime.String(),
+		"end_time":   endTime.String(),
+		"duration":   totalDuration.String(),
+		"n":          len(j.events),
+	})
+
+	return catcher.Resolve()
+}
+
+func (j *eventMetaJob) dispatch(notifications []notification.Notification) error {
+	catcher := grip.NewSimpleCatcher()
+	for i := range notifications {
+		if notificationIsEnabled(j.flags, &notifications[i]) {
+			catcher.Add(j.q.Put(newEventNotificationJob(notifications[i].ID)))
+
+		} else {
+			catcher.Add(notifications[i].MarkError(errors.New("sender disabled")))
+		}
+	}
+
+	return catcher.Resolve()
+}
+
+func (j *eventMetaJob) Run(_ context.Context) {
+	defer j.MarkComplete()
+
+	if j.q == nil {
+		env := evergreen.GetEnvironment()
+		j.q = env.LocalQueue()
```

Comment:
we probably don't want these in the local queue?

I think we may have discussed our logic here, but a comment might be in order. 

it might also be the case that the senders themselves will be buffered and rate limited, and so just calling "send" might be enough and we can remove the nested queue.

Reply:
Nope, it was supposed to be RemoteQueue(). We're going to have to do that with slack, but I have a ticket open for that in the future.